providers/unleash/src/test/java/dev/openfeature/contrib/providers/unleash/UnleashProviderTest.java

```diff
@@ -0,0 +1,293 @@
+import java.util.List;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * UnleashProvider test, based on APIs mocking.
+ * Inspired by Unleash tests.
+ */
+@WireMockTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class UnleashProviderTest {
+
+    public static final String FLAG_NAME = "variant-flag";
+    public static final String VARIANT_FLAG_NAME = "variant-flag";
+    public static final String VARIANT_FLAG_VALUE = "v1";
+    public static final String USERS_FLAG_NAME = "users-flag";
+    public static final String JSON_VARIANT_FLAG_NAME = "json-flag";
+    public static final String JSON_VARIANT_FLAG_VALUE = "{ a: 1 }";
+    public static final String CSV_VARIANT_FLAG_NAME = "csv-flag";
+    public static final String CSV_VARIANT_FLAG_VALUE = "a,b,c";
+    private static UnleashProvider unleashProvider;
+    private static Client client;
+
+    @BeforeAll
+    void setUp(WireMockRuntimeInfo wmRuntimeInfo) {
+        stubFor(any(anyUrl()).willReturn(aResponse()
+            .withStatus(200)
+            .withBody("{}")));
+        String unleashAPI = "http://localhost:" + wmRuntimeInfo.getHttpPort() + "/api/";
+        String backupFileContent = readBackupFile();
+        mockUnleashAPI(backupFileContent);
+        unleashProvider = buildUnleashProvider(true, unleashAPI, backupFileContent, new TestSubscriber());
+        OpenFeatureAPI.getInstance().setProviderAndWait("sync", unleashProvider);
+        client = OpenFeatureAPI.getInstance().getClient("sync");
+    }
+
+    @AfterAll
+    public void shutdown() {
+        unleashProvider.shutdown();
+    }
+
+    private void mockUnleashAPI(String backupFileContent) {
+        stubFor(
+            get(urlEqualTo("/api/client/features"))
+                .withHeader("Accept", equalTo("application/json"))
+                .willReturn(
+                    aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(backupFileContent)));
+        stubFor(post(urlEqualTo("/api/client/register")).willReturn(aResponse().withStatus(200)));
+    }
+
+    @SneakyThrows
+    private UnleashProvider buildUnleashProvider(boolean synchronousFetchOnInitialisation, String unleashAPI, String backupFileContent, TestSubscriber testSubscriber) {
```

Comment:
I don't think `backupFileContent` is ever used here. Doesn't look like it affects the tests but it might be worth taking out

Reply:
it is needed, similar to tests at Unleash.

Reply:
I'm also confused about this :sweat_smile:

Reply:
I pulled the branch and removed this param, and everything seems to work the same.

Can you clarify @liran2000 ?

Reply:
Removed